### PR TITLE
Core/ICC: Gossip DB Teleport

### PR DESCRIPTION
**Pull request by ghost**
Hello, please do not criticize I'm just learning. I noticed that you do
not use teleport gossip in the citadel. I tried to do it, it works well.
But all I did was ask for help.

### CODE REVIEW

File: sql/updates/world/2016_99_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+DELETE FROM gossip_menu_option WHERE menu_id=11017;
+INSERT INTO `gossip_menu_option` (`menu_id`, `id`, `option_text`, `OptionBroadcastTextID`, `option_id`, `npc_option_npcflag`) VALUES 
+(11017, 0, 'Teleport to Light\'s Hammer.', 37671, 1, 1),
+(11017, 1, 'Teleport to the Oratory of the Damned.', 37722, 1, 1),
+(11017, 2, 'Teleport to the Rampart of Skulls.', 37723, 1, 1),
+(11017, 3, 'Teleport to the Deathbringer\'s Rise.', 37724, 1, 1),
+(11017, 4, 'Teleport to the Upper Spire.', 37725, 1, 1),
+(11017, 5, 'Teleport to Sindragosa\'s Lair.', 37728, 1, 1);
```
Comment (joschiwald) [inline sql/updates/world/2016_99_99_99_world.sql]: i found this in my gossip sniff file

```sql
INSERT INTO `gossip_menu` (`entry`, `text_id`) VALUES
(11013, 15221), -- 202242
(11014, 15221), -- 202245
(11016, 15221), -- 202244
(11017, 15221), -- 202235

INSERT INTO `gossip_menu_option` (`menu_id`, `id`, `option_icon`, `option_text`, `box_coded`, `box_money`, `box_text`) VALUES
(11013, 1, 0, 'Teleport to the Oratory of the Damned.', 0, 0, ''), -- 202242
(11013, 3, 0, 'Teleport to the Rampart of Skulls.', 0, 0, ''), -- 202242
(11013, 4, 0, 'Teleport to the Deathbringer''s Rise.', 0, 0, ''), -- 202242
(11013, 5, 0, 'Teleport to the Upper Spire.', 0, 0, ''), -- 202242

(11014, 0, 0, 'Teleport to Light''s Hammer.', 0, 0, ''), -- 202245
(11014, 3, 0, 'Teleport to the Rampart of Skulls.', 0, 0, ''), -- 202245
(11014, 4, 0, 'Teleport to the Deathbringer''s Rise.', 0, 0, ''), -- 202245
(11014, 5, 0, 'Teleport to the Upper Spire.', 0, 0, ''), -- 202245

(11016, 0, 0, 'Teleport to Light''s Hammer.', 0, 0, ''), -- 202244
(11016, 1, 0, 'Teleport to the Oratory of the Damned.', 0, 0, ''), -- 202244
(11016, 3, 0, 'Teleport to the Rampart of Skulls.', 0, 0, ''), -- 202244
(11016, 5, 0, 'Teleport to the Upper Spire.', 0, 0, ''), -- 202244

(11017, 0, 0, 'Teleport to Light''s Hammer.', 0, 0, ''), -- 202235
(11017, 1, 0, 'Teleport to the Oratory of the Damned.', 0, 0, ''), -- 202235
(11017, 3, 0, 'Teleport to the Rampart of Skulls.', 0, 0, ''), -- 202235
(11017, 4, 0, 'Teleport to the Deathbringer''s Rise.', 0, 0, ''), -- 202235
```

Comment (DevRival) [inline sql/updates/world/2016_99_99_99_world.sql]: I know that's why I ask the help. As you can see there are different going and what they are.

Comment (joschiwald) [inline sql/updates/world/2016_99_99_99_world.sql]: each teleporter has its own gossip, sadly i don't have sniffs of rampart of skulls and sindragosas lair teleporter

Comment (joschiwald) [inline sql/updates/world/2016_99_99_99_world.sql]: by aokromes
```sql
DELETE FROM gossip_menu WHERE `entry` IN (11013, 11014, 11015, 11016, 11017, 11018);
INSERT INTO gossip_menu (`entry`, `text_id`) VALUES
(11013, 15221), -- 11013
(11014, 15221), -- 11014
(11015, 15221), -- 11015
(11016, 15221), -- 11016
(11017, 15221), -- 11017
(11018, 15221); -- 11018

DELETE FROM gossip_menu_option WHERE `menu_id` IN (11013, 11014, 11015, 11016, 11017, 11018);
INSERT INTO gossip_menu_option (`menu_id`, `id`, `option_icon`, `option_text`, `OptionBroadcastTextID`, `option_id`, `npc_option_npcflag`, `action_menu_id`, `action_poi_id`, `box_coded`, `box_money`, `box_text`, `BoxBroadcastTextID`) VALUES
(11013, 6, 0, 'Teleport to Sindragosa''s Lair.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11013
(11013, 5, 0, 'Teleport to the Upper Spire.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11013
(11013, 4, 0, 'Teleport to the Deathbringer''s Rise.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11013
(11013, 3, 0, 'Teleport to the Rampart of Skulls.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11013
(11013, 1, 0, 'Teleport to the Oratory of the Damned.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11013

(11014, 6, 0, 'Teleport to Sindragosa''s Lair.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11014
(11014, 5, 0, 'Teleport to the Upper Spire.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11014
(11014, 4, 0, 'Teleport to the Deathbringer''s Rise.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11014
(11014, 3, 0, 'Teleport to the Rampart of Skulls.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11014
(11014, 0, 0, 'Teleport to Light''s Hammer.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11014

(11015, 6, 0, 'Teleport to Sindragosa''s Lair.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11015
(11015, 5, 0, 'Teleport to the Upper Spire.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11015
(11015, 4, 0, 'Teleport to the Deathbringer''s Rise.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11015
(11015, 1, 0, 'Teleport to the Oratory of the Damned.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11015
(11015, 0, 0, 'Teleport to Light''s Hammer.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11015

(11016, 6, 0, 'Teleport to Sindragosa''s Lair.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11016
(11016, 5, 0, 'Teleport to the Upper Spire.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11016
(11016, 3, 0, 'Teleport to the Rampart of Skulls.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11016
(11016, 1, 0, 'Teleport to the Oratory of the Damned.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11016
(11016, 0, 0, 'Teleport to Light''s Hammer.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11016

(11017, 6, 0, 'Teleport to Sindragosa''s Lair.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11017
(11017, 4, 0, 'Teleport to the Deathbringer''s Rise.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11017
(11017, 3, 0, 'Teleport to the Rampart of Skulls.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11017
(11017, 1, 0, 'Teleport to the Oratory of the Damned.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11017
(11017, 0, 0, 'Teleport to Light''s Hammer.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11017

(11018, 5, 0, 'Teleport to the Upper Spire.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11018
(11018, 4, 0, 'Teleport to the Deathbringer''s Rise.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11018
(11018, 3, 0, 'Teleport to the Rampart of Skulls.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11018
(11018, 1, 0, 'Teleport to the Oratory of the Damned.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 11018
(11018, 0, 0, 'Teleport to Light''s Hammer.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN); -- 11018
```

**Comment by DevRival** [comment]
Yes, I know that, but we have a little problem. All these have id missing, and that's why I made a new entry in the Gossip.

**Comment by DevRival** [comment]
You can delete all the unused id, and to use only one.

**Comment by joschiwald** [comment]
what do you mean? the sql i pasted is from sniff, everything different is wrong like your sql

**Comment by DevRival** [comment]
I understand what I did wrong?

**Comment by mthsena** [comment]
@DevRival 
Yours code is not exactly wrong, I guess that they are trying to say that all need to be like the sniffs and another method is considered wrong.

**Comment by tkrokli** [comment]
@DevRival : just to fill in the blanks in Aokromes' SQL (and sort the lines in ascending order):
```sql
DELETE FROM gossip_menu WHERE `entry` IN (11013, 11014, 11015, 11016, 11017, 11018);
INSERT INTO gossip_menu (`entry`, `text_id`) VALUES
(11013, 15221), -- 11013
(11014, 15221), -- 11014
(11015, 15221), -- 11015
(11016, 15221), -- 11016
(11017, 15221), -- 11017
(11018, 15221); -- 11018

DELETE FROM gossip_menu_option WHERE `menu_id` IN (11013, 11014, 11015, 11016, 11017, 11018);
INSERT INTO gossip_menu_option (`menu_id`, `id`, `option_icon`, `option_text`, `OptionBroadcastTextID`, `option_id`, `npc_option_npcflag`, `action_menu_id`, `action_poi_id`, `box_coded`, `box_money`, `box_text`, `BoxBroadcastTextID`) VALUES
(11013, 1, 0, 'Teleport to the Oratory of the Damned.', 37722, 1, 1, 0, 0, 0, 0, '', 0), -- Light's Hammer
(11013, 3, 0, 'Teleport to the Rampart of Skulls.',     37723, 1, 1, 0, 0, 0, 0, '', 0), -- Light's Hammer
(11013, 4, 0, 'Teleport to the Deathbringer''s Rise.',  37724, 1, 1, 0, 0, 0, 0, '', 0), -- Light's Hammer
(11013, 5, 0, 'Teleport to the Upper Spire.',           37725, 1, 1, 0, 0, 0, 0, '', 0), -- Light's Hammer
(11013, 6, 0, 'Teleport to Sindragosa''s Lair.',        37728, 1, 1, 0, 0, 0, 0, '', 0), -- Light's Hammer

(11014, 0, 0, 'Teleport to Light''s Hammer.',           37671, 1, 1, 0, 0, 0, 0, '', 0), -- Oratory of the Damned
(11014, 3, 0, 'Teleport to the Rampart of Skulls.',     37723, 1, 1, 0, 0, 0, 0, '', 0), -- Oratory of the Damned
(11014, 4, 0, 'Teleport to the Deathbringer''s Rise.',  37724, 1, 1, 0, 0, 0, 0, '', 0), -- Oratory of the Damned
(11014, 5, 0, 'Teleport to the Upper Spire.',           37725, 1, 1, 0, 0, 0, 0, '', 0), -- Oratory of the Damned
(11014, 6, 0, 'Teleport to Sindragosa''s Lair.',        37728, 1, 1, 0, 0, 0, 0, '', 0), -- Oratory of the Damned

(11015, 0, 0, 'Teleport to Light''s Hammer.',           37671, 1, 1, 0, 0, 0, 0, '', 0), -- Rampart of Skulls
(11015, 1, 0, 'Teleport to the Oratory of the Damned.', 37722, 1, 1, 0, 0, 0, 0, '', 0), -- Rampart of Skulls
(11015, 4, 0, 'Teleport to the Deathbringer''s Rise.',  37724, 1, 1, 0, 0, 0, 0, '', 0), -- Rampart of Skulls
(11015, 5, 0, 'Teleport to the Upper Spire.',           37725, 1, 1, 0, 0, 0, 0, '', 0), -- Rampart of Skulls
(11015, 6, 0, 'Teleport to Sindragosa''s Lair.',        37728, 1, 1, 0, 0, 0, 0, '', 0), -- Rampart of Skulls

(11016, 0, 0, 'Teleport to Light''s Hammer.',           37671, 1, 1, 0, 0, 0, 0, '', 0), -- Deathbringer's Rise
(11016, 1, 0, 'Teleport to the Oratory of the Damned.', 37722, 1, 1, 0, 0, 0, 0, '', 0), -- Deathbringer's Rise
(11016, 3, 0, 'Teleport to the Rampart of Skulls.',     37723, 1, 1, 0, 0, 0, 0, '', 0), -- Deathbringer's Rise
(11016, 5, 0, 'Teleport to the Upper Spire.',           37725, 1, 1, 0, 0, 0, 0, '', 0), -- Deathbringer's Rise
(11016, 6, 0, 'Teleport to Sindragosa''s Lair.',        37728, 1, 1, 0, 0, 0, 0, '', 0), -- Deathbringer's Rise

(11017, 0, 0, 'Teleport to Light''s Hammer.',           37671, 1, 1, 0, 0, 0, 0, '', 0), -- Upper Spire
(11017, 1, 0, 'Teleport to the Oratory of the Damned.', 37722, 1, 1, 0, 0, 0, 0, '', 0), -- Upper Spire
(11017, 3, 0, 'Teleport to the Rampart of Skulls.',     37723, 1, 1, 0, 0, 0, 0, '', 0), -- Upper Spire
(11017, 4, 0, 'Teleport to the Deathbringer''s Rise.',  37724, 1, 1, 0, 0, 0, 0, '', 0), -- Upper Spire
(11017, 6, 0, 'Teleport to Sindragosa''s Lair.',        37728, 1, 1, 0, 0, 0, 0, '', 0), -- Upper Spire

(11018, 0, 0, 'Teleport to Light''s Hammer.',           37671, 1, 1, 0, 0, 0, 0, '', 0), -- Sindragosa's Lair
(11018, 1, 0, 'Teleport to the Oratory of the Damned.', 37722, 1, 1, 0, 0, 0, 0, '', 0), -- Sindragosa's Lair
(11018, 3, 0, 'Teleport to the Rampart of Skulls.',     37723, 1, 1, 0, 0, 0, 0, '', 0), -- Sindragosa's Lair
(11018, 4, 0, 'Teleport to the Deathbringer''s Rise.',  37724, 1, 1, 0, 0, 0, 0, '', 0), -- Sindragosa's Lair
(11018, 5, 0, 'Teleport to the Upper Spire.',           37725, 1, 1, 0, 0, 0, 0, '', 0); -- Sindragosa's Lair
```

**Comment by joschiwald** [comment]
the ongossiphello code must be converted to conditions, then you need to adjust the ongossipselect code (move to sgossipselect)

**Comment by DevRival** [comment]
I'll close the pull until better times.

File: sql/updates/world/2016_99_99_99_world.sql (hunk 1 above)
Comment (Aokromes) [inline sql/updates/world/2016_99_99_99_world.sql]: the 1st part is already on db, the rest have 11015 and 11018 missing and some rows missing.

**Comment by joschiwald** [comment]
ea38a7d95bb7d67005c7de1a924d288561b0da13

**Pull request closed without merging**